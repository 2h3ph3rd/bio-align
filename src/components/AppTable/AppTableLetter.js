import * as React from 'react'
import Typography from '@mui/material/Typography'
import { blueGrey } from '@mui/material/colors'

import AppTableCard from './AppTableCard'

export default function AppTableLetter(props) {
  return (
    <AppTableCard backgroundColor={blueGrey['50']}>
      <Typography sx={{ fontWeight: 'bold' }}>{props.letter}</Typography>
    </AppTableCard>
  )
}
