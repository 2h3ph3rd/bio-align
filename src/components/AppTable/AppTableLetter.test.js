import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import AppTableLetter from './AppTableLetter'

const fakeLetter = 'A'

it('renders correctly', () => {
  const tree = renderer
    .create(<AppTableLetter letter={fakeLetter}></AppTableLetter>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render the letter given', () => {
  render(<AppTableLetter letter={fakeLetter}></AppTableLetter>)

  const div = screen.getByText(fakeLetter)
  expect(div).toHaveTextContent(fakeLetter)
})
