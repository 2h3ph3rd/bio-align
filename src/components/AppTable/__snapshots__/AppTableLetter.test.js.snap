// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1uj8utb-MuiPaper-root"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
  >
    A
  </p>
</div>
`;
